pkg/controller/servicebindingrequest/secret.go

```diff
@@ -47,14 +49,26 @@ func (s *secret) createOrUpdate(payload map[string][]byte) (*unstructured.Unstru
 
 	logger.Debug("Attempt to create secret...")
 	_, err = resourceClient.Create(u, metav1.CreateOptions{})
-	if err != nil && !errors.IsAlreadyExists(err) {
-		return nil, err
-	}
-
-	logger.Debug("Secret already exists, updating contents instead...")
-	_, err = resourceClient.Update(u, metav1.UpdateOptions{})
 	if err != nil {
-		return nil, err
+		if errors.IsAlreadyExists(err) {
```

Comment:
I believe the right flow would be:

* Get the resource
* If not found, create
* If found and existing data is different from the given payload, update the secret.